api/src/main/java/org/datacleaner/configuration/RemoteServerData.java

```diff
@@ -33,4 +33,8 @@
     public String getUsername();
 
     public String getPassword();
+
+    public void setUsername(String username);
```

Comment:
I did really like that this was immutable... It gives a lot of guarantees that mutability breaks.

So to explain a bit of background: Normally when you have an immutable structure, you don't try to mutate the existing objects - you rather just create a NEW object that represents the change.